metaflow/plugins/aip/argo_utils.py

```diff
@@ -12,155 +16,299 @@
 logger = _get_aip_logger()
 
 
-def run_argo_workflow(
-    kubernetes_namespace: str,
-    template_name: str,
-    parameters: Optional[dict] = None,
-    wait_timeout: Union[int, float, datetime.timedelta] = 0,
-    **kwarg,  # Other parameters for wait function
-) -> Tuple[str, str]:
-    try:
-        # TODO(talebz): add tag of origin-run-id to correlate parent flow
-        workflow_manifest: Dict[str, Any] = ArgoClient(
-            namespace=kubernetes_namespace,
-        ).trigger_workflow_template(template_name, parameters)
-    except Exception as e:
-        raise AIPException(str(e))
-
-    argo_run_id = workflow_manifest["metadata"]["name"]
-    argo_run_uid = workflow_manifest["metadata"]["uid"]
-
-    if wait_timeout:  # int, float and datetime.timedelta all evaluates to False when 0
-        wait_for_argo_run_completion(
-            run_id=argo_run_id,
-            kubernetes_namespace=kubernetes_namespace,
-            wait_timeout=wait_timeout,
-            **kwarg,
-        )
-
-    return argo_run_id, argo_run_uid
+class ArgoHelper:
+    def __init__(self, kubernetes_namespace: str = KUBERNETES_NAMESPACE):
+        """
+        Args:
+            kubernetes_namespace: Namespace where Argo is running.
+                Required as the defaults provided in the ArgoClient is usually not what customers desire.
+                TODO: This namespace can be default to the current namespace if the script is ran within a cluster.
+        """
+        self._client = ArgoClient(namespace=kubernetes_namespace)
+
+    def trigger(
```

Comment:
thank you for the techreview! from (@cavandervoort feedback) : in v1 `run_argo_workflow(template_name=)` would run the latest, but is now `trigger(template_name=)` which runs the exact or specific workflow template.  This could confuse users.  

Could you please rename this to `trigger_specific(template_name=)` and add documentation to both functions on the ZG template naming with example names and invocations to these functions?